front end template completed

felipeaocampo opened the pull request:
Created the three main sections for our application: info panel, nav, and canvas. Info panel has form which uploads and sends uploaded file
<img width="1512" alt="for nattie" src="https://user-images.githubusercontent.com/95617522/222934782-0d9553d4-ca0a-42cf-bfa9-9d22b40823a0.png">

felipeaocampo committed:
front end template completed

nattiechan commented on client/components/Info Panel/InfoPanel.js:
```diff
@@ -0,0 +1,41 @@
+        accusamus provident nostrum sequi quas!
+      </p>
+      <form
+        action="/test"
```
Does this API endpoint work currently? I assume not? If so we may want to open a task in Asana to connect the front and back end to make sure we do not miss this.

felipeaocampo replied:
Hey Nattie, No. This endpoint is a dummy endpoint. In order for it to work so the backend team can use it to send themselves the file through the front end, I'd need to add a few lines of code. I can do this tonight.

nattiechan commented on client/components/Nav/Nav.js:
```diff
@@ -0,0 +1,14 @@
+const Nav = () => {
+  return (
+    <nav className="nav">
+      <h3 className="nav__script">Script</h3>
+      <h3 className="nav__diagram">ER Diagram</h3>
```
Going `id` instead of `className`, imo may be better. Down the road if we want to add tests, finding an element by ID is much easier than class names.

nattiechan replied:
Also, do we have a strategy for switching between the script and ER diagram? It's fine that you don't have this implemented right now but I just want to make sure we have a strategy

felipeaocampo replied:
Yes :). The strategy is simple. Use a useState hook to determine when to add a special class "active". Frontend will create a state var called activeTabScript and declare it to false. Then in the JSX when this variable is false, we will conditionally render a class named "active" which just has the same border bottom that appears when the elements are hovered. Then we need to set up two click handlers, one for script and another for diagram. When script is clicked, activeTabScript is set to true. When diagram is clicked, it is set to false.

nattiechan commented on client/components/Info Panel/InfoPanel.js:
```diff
@@ -0,0 +1,41 @@
+const InfoPanel = () => {
+  const uploadExcelHandler = (e) => {
+    e.preventDefault();
+    console.log(`File was sent!`);
```
A TODO here (for future iterations) maybe is to have a text that would be displayed on front end (and then hidden afterwards) instead of console logging.

Also, we may also want to alert the user if the upload fails.

felipeaocampo replied:
Oh for sure! Adding that on the task list

nattiechan reviewed (commented):
Looks great so far! I love the screenshot. Some thoughts below.

felipeaocampo committed:
Merge branch 'dev' of https://github.com/ExcQL/ExcQL into front-end

felipeaocampo committed:
added frontend functionality for sending file to backend endpoint, clicking and showing active tab, and toggling between script and diagram components on click

felipeaocampo commented:
Hey, Nattie. I added the functionality we were talking about and basically completed the front end now as complete as it can be without the diagram or script portion, and the actual paragraph description text. I added, committed, and pushed by code. I tried to open another pull request, but gitHub is telling me that this one is still open. Should I close it and resent another pull request for the updated frontend code, or is there a way for you to see the current state of the code I just pushed?

nattiechan reviewed (approved):
It looks fine for now. Let's merge this into dev so you can continue working on the front end tasks as you please!

nattiechan commented:
> Hey, Nattie. I added the functionality we were talking about and basically completed the front end now as complete as it can be without the diagram or script portion, and the actual paragraph description text. I added, committed, and pushed by code. I tried to open another pull request, but gitHub is telling me that this one is still open. Should I close it and resent another pull request for the updated frontend code, or is there a way for you to see the current state of the code I just pushed?

Usually if your branch has a PR open, any changes after the PR opened will automatically appear here. So there is no need to open a second PR. Since the request is approved, how you can click "merge pull request" below as long as there are no conflicts!
